stdlib/tkinter/__init__.pyi

```diff
@@ -1112,17 +1115,164 @@ class Canvas(Widget, XView, YView):
+        fill: _Color = ...,
+        joinstyle: Literal["round", "bevel", "miter"] = ...,
+        offset: Union[int, str] = ...,
+        smooth: Literal[0, 1] = ...,
+        splinesteps: int = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, tuple[str, ...]] = ...,
+        width: Union[int, str] = ...,
+    ) -> _CanvasObjectId: ...
+    def create_oval(
+        self,
+        __x0: int,
+        __y0: int,
+        __x1: int,
+        __y1: int,
+        activedash: _Color = ...,
+        activefill: _Color = ...,
+        activeoutline: _Color = ...,
+        activeoutlinestipple: _Color = ...,
+        activestipple: str = ...,
+        activewidth: Union[int, str] = ...,
+        dash: Union[tuple[int], tuple[int, int], tuple[int, int, int, int]] = ...,
+        dashoffset: Union[int, str] = ...,
+        disableddash: _Color = ...,
+        disabledfill: _Color = ...,
+        disabledoutline: _Color = ...,
+        disabledoutlinestipple: _Color = ...,
+        disabledstipple: _Bitmap = ...,
+        disabledwidth: Union[int, str] = ...,
+        fill: _Color = ...,
+        offset: Union[int, str] = ...,
+        outline: _Color = ...,
+        outlineoffset: Union[int, str] = ...,
+        outlinestipple: _Bitmap = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, tuple[str, ...]] = ...,
+        width: Union[int, str] = ...,
+    ) -> _CanvasObjectId: ...
+    def create_polygon(
+        self,
+        __x0: int,
+        __y0: int,
+        __x1: int,
+        __y1: int,
+        *args: int,
+        activedash: _Color = ...,
+        activefill: _Color = ...,
+        activeoutline: _Color = ...,
+        activeoutlinestipple: _Color = ...,
+        activestipple: str = ...,
+        activewidth: Union[int, str] = ...,
+        dash: Union[tuple[int], tuple[int, int], tuple[int, int, int, int]] = ...,
+        dashoffset: Union[int, str] = ...,
+        disableddash: _Color = ...,
+        disabledfill: _Color = ...,
+        disabledoutline: _Color = ...,
+        disabledoutlinestipple: _Color = ...,
+        disabledstipple: _Bitmap = ...,
+        disabledwidth: Union[int, str] = ...,
+        fill: _Color = ...,
+        joinstyle: Literal["round", "bevel", "miter"] = ...,
+        offset: Union[int, str] = ...,
+        outline: _Color = ...,
+        outlineoffset: Union[int, str] = ...,
+        outlinestipple: _Bitmap = ...,
+        smooth: Literal[0, 1] = ...,
+        splinesteps: int = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, tuple[str, ...]] = ...,
+        width: Union[int, str] = ...,
+    ) -> _CanvasObjectId: ...
+    def create_rectangle(
+        self,
+        __x0: int,
+        __y0: int,
+        __x1: int,
+        __y1: int,
+        activedash: _Color = ...,
```

Comment:
A few things that also apply to other `create_foo()` methods:
- The coordinates can be `float`. Use `float` instead of `int`. There's no need to do `Union[float, int]`, because type checkers consider `int` to also be included when you say `float`.
- Use `*` to mark rest of the arguments keyword-only: `__y1: int, *, activedash: _Color = ...`

Reply:
Okay, I see my mistake. Both my original and current revisions were incorrect because they considered the latter arguments could be positional.

Reply:
A lot of the coordinates are currently `Union[int, str]` - so I assume those should become `Union[float, str]`.

Reply:
You can also try setting the options to e.g. `"2c"`. If that works, the correct type is `_ScreenUnits`.